synchronization/src/main/java/de/cyface/synchronization/HttpConnection.java

```diff
@@ -161,15 +161,17 @@ public HttpResponse post(@NonNull final HttpURLConnection connection, final @Non
             throws SynchronisationException, ResponseParsingException, BadRequestException, UnauthorizedException {
 
         // Generate header
-        final long filePartSize = transferTempFile.length() + TAIL.length();
-        Validate.isTrue(filePartSize > 0);
-        final String header = generateHeader(filePartSize, metaData, fileName);
-
-        // Set content length (count the bytes not the string length!)
-        final long requestLength = header.getBytes().length + filePartSize;
-        // This should be obsolete with setFixedLengthStreamingMode:
-        // connection.setRequestProperty("Content-length", String.valueOf(requestLength));
-        connection.setFixedLengthStreamingMode((int)requestLength);
+        // Attention: Parts of the header (Content-Type, boundary, request method, user agent) are already set
+        final String remainingHeader = generateHeader(metaData, fileName);
+
+        // Set the fixed number of bytes which will be written to the OutputStream
+        final long binarySize = transferTempFile.length();
+        final long bytesWrittenToOutputStream = calculateBytesWrittenToOutputStream(remainingHeader, binarySize);
```

Comment:
I have this in a method to test this in a unit test to make sure we don't break it again.